### ISSUE TITLE
Hive iOS - "Whoops! Request timeout."

### ISSUE THREAD

**Issue by KudoJin** [hivewallet/hive-ios#47]
So, I've had this issue for months now and now it's getting ridiculous.

When I first got the HIVE iOS app it worked perfectly, but since 3-4 months ago I've been having this issue where if I try to access my LTC wallet I get a "Woops! Request timeout." error and it won't let me access my LTC wallet. I can access the wallet, but only after about 20 different attempts at accessing the wallet.

I can access the BTC side with zero issues, it is only the LTC wallet that is giving me this issue.

It is not the Internet connection as I have tried this on about 5 different networks, some of which I know are alright, and I still get this issue.

I have tried reinstalling the app and I still get this issue.

I have even since replaced the iPhone (my old phone was breaking down anyway) and I STILL GET THE ISSUE.

Is there any way to fix this? It's getting really annoying and if I can't fix it I may even switch over to a different wallet.

![fr_464](https://cloud.githubusercontent.com/assets/12526353/8123511/9569dfa0-1066-11e5-8dff-3e1fc7b08bde.jpg)

**Comment by KudoJin** [comment]
Is there nobody who can help me with this issue?

If not, I'll be switching to ZiftrWallet once it gets released for iOS.

**Comment by ghost** [comment]
This is what @weilu says:

`Caching API responses in the browser partially solves the problem for frequent users -- it works until user clears the browser cache, or logs in the first time (or after a long time) using another browser. Not to mention that caching isn't trivial to implement correctly. This is among those _any_ solutions but not a _good_ one.

There are smart ways to defer making requests to the blockchain API:
1. Open the wallet as soon as all the used addresses are discovered. At this point we have the balance. User can see the balance, but not the transaction history, and they will not be able to spend any money.
2. Fetch all the unspents for the discovered addresses. After this is done, user will be able to send money.
3. Load transactions. Given the limitation of blockr, we cannot query transactions by block height for a given set of addresses, so there's no sure way to load the latest transactions first. Our best bet is to load the transactions in order of the newest discovered to the first discovered addresses. In UX words, 1) it's not possible to load the latest n transactions first, then wait for user to scroll to load more, while guarantee correctness. 2) if we decide to show transactions as they come in, transactions will come in no particular order -- you will see transactions inserted in between already loaded transactions, if we want to preserve the current reverse chronological order in the history tab. Whether we want to show transactions as they come in or show nothing until all transactions are loaded, that's a UX decision I'll leave to Matt.
4. Once all above is done, we can consider caching transactions to speed up step 3. The number of confirmations can be updated by caching each transaction's block height, and then finding out the current block height. We need to be careful about unconfirmed transactions -- I'd advice to not cache any transactions with less than 6 confirmations, but as we all know, 6 isn't a magic number when the amount involved is big, but it's probably a good enough start.`

**Comment by ghost** [comment]
tl;dr 

The browser and/or API resets the connection after some timeout. You can only get into your wallet if all your requests manage to finish before the timeout.

An implemented solution will take some time, because at the moment no one is working on Hive full-time. This will change next month, but you probably don't want to wait that long.

**Comment by ghost** [comment]
Can you guys please try https://hive-js.herokuapp.com ? Wei has been reducing the payload to reduce these issues, which appear to be entirely timing-based.

**Comment by ghost** [comment]
Closing this and other redundant issues now; #237 and #239 exist to consolidate. You guys may continue to comment here however.
